src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -1037,13 +1250,476 @@ subroutine get_lowmode_loss(i,j,G,CS,mechanism,TKE_loss_sum)
+        zdepth = zdepth + 0.5*dz(i,k)
+      endif
+    enddo
+
+    if (G%mask2dT(i,j) > 0.0) then
+      ! allow for difference less than verification threshold
+      renorm_StLau = 1.0
+      renorm_StLau_slope = 1.0
+      if (abs(tmp_StLau -1.0) > threshold_verif) renorm_StLau = 1.0 / tmp_StLau
+      if (abs(tmp_StLau_slope -1.0) > threshold_verif) renorm_StLau_slope = 1.0 / tmp_StLau_slope
+
+      do k=1,nz
+        profile_StLaurent(k) = profile_StLaurent(k) * renorm_StLau
+        profile_StLaurent_slope(k) = profile_StLaurent_slope(k) * renorm_StLau_slope
+      enddo
+    endif
+
+    ! verif integrals
+    if (CS%debug) then
+      if (G%mask2dT(i,j) > 0.0) then
+         verif_N = 0.
+         verif_N2 = 0.
+         verif_bbl = 0.
+         verif_stl1 = 0.
+         verif_stl2 = 0.
+         do k=1,nz
+           verif_N = verif_N + (profile_N(k) * dz(i,k))
+           verif_N2 = verif_N2 + (profile_N2(k) * dz(i,k))
+           verif_bbl = verif_bbl + (profile_BBL(k) * dz(i,k))
+           verif_stl1 = verif_stl1 + (profile_StLaurent(k) * dz(i,k))
+           verif_stl2 = verif_stl2 + (profile_StLaurent_slope(k) * dz(i,k))
+         enddo
+
+         if (abs(verif_N -1.0) > threshold_verif) then
+           print *, i, j, verif_N
+           call MOM_error(FATAL, "mismatch integral for N profile")
+         endif
+         if (abs(verif_N2 -1.0) > threshold_verif) then
+           print *, i, j, verif_N2
+           call MOM_error(FATAL, "mismatch integral for N2 profile")
+         endif
+         if (abs(verif_bbl -1.0) > threshold_verif) then
+           print *, i, j, verif_bbl
+           call MOM_error(FATAL, "mismatch integral for bbl profile")
+         endif
+         if (abs(verif_stl1 -1.0) > threshold_verif) then
+           print *, i, j, verif_stl1
+           call MOM_error(FATAL, "mismatch integral for stl1 profile")
+         endif
+         if (abs(verif_stl2 -1.0) > threshold_verif) then
+           print *, i, j, verif_stl2
+           call MOM_error(FATAL, "mismatch integral for stl2 profile")
+         endif
+
+      endif
+    endif
+
+    ! note on units: TKE_to_Kd = 1 / ((g/rho0) * drho) Z-1 T2
+    ! mult by dz gives -1/N2 in T2
+
+    ! get TKE loss value and compute diffusivites in layers
+    if (CS%apply_background_drag) then
+      call get_lowmode_loss(i, j, G, CS, "LeakDrag", TKE_loss)
+      ! insert logic to switch between profiles here
+      ! if trim(CS%leak_profile) == "N2" then
+      profile_leak(i,:) = profile_N2(:)
+      ! elseif trim(CS%leak_profile) == "N" then
+      ! profile_leak(:) = profile_N(:)
+      ! something else
+      ! endif
+      Kd_leak_lay(:) = 0.
+      do k=1,nz
+        ! layer diffusivity for processus
+        if (dz(i,k) >= CS%min_depth_layer_Kd) then
+          TKE_to_Kd_lim = min(TKE_to_Kd(i,k), CS%max_TKE_to_Kd)
+          if (non_Bous) then
+            Kd_leak_lay(k) = GV%Z_to_H * TKE_loss * TKE_to_Kd_lim * profile_leak(i,k) * dz(i,k) * tv%SpV_avg(i,j,k)
```

Comment:
The expression on line 1558 (and elsewhere on lines 1560, 1584, 1586, 1610, 1612, 1636, 1638, 1662 and 1664) would not need GV%Z_to_H if it were using `h(i,k)` in place of `dz(i,k)`.  Additionally, we should perhaps recast the various variables like `profile_leak` to use units of `[H-1 ~> m-1 or m2 kg-1]` and TKE_loss to be in units of `[H Z2 T-3 ~> m3 s-3 or W m-2]`, as is done elsewhere in MOM_set_diffusivity.  All of these changes would avoid introducing the Boussinesq reference density into non-Boussinesq calculations.

Also, instead of dividing by GV%Rho0, it is equivalent to multiply by `GV%H_to_Z * GV%RZ_to_H` (for either Boussinesq or non-Boussinesq mode) and then you could rearrange and replace `GV%Z_to_H * GV%H_to_Z` with 1 or just eliminate it.

(If we keep this code as is, there will probably be a second pull request soon to do the non-Boussinesq conversion of this code.  This would be answer preserving for Boussinesq configurations.)

Reply:
I've replaced GV%Rho0 by GV%H_to_Z * GV%RZ_to_H

Not sure how to elegantly brings the profiles, which are either defined as a distance to the bottom or normalized by the total ocean depth, back in [H-1] since the `dz_to_thickness` functions are not elemental.